ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/StreamedQueryResourceTest.java

```diff
@@ -66,69 +69,83 @@
 
-  @Test
-  public void shouldReturn400OnBadStatement() throws Exception {
-    final String queryString = "SELECT * FROM test_stream;";
-
-    final KsqlConfig ksqlConfig = mock(KsqlConfig.class);
-    final KsqlEngine mockKsqlEngine = mock(KsqlEngine.class);
-    final KafkaTopicClient mockKafkaTopicClient = mock(KafkaTopicClientImpl.class);
+  @Mock(MockType.NICE)
+  private KsqlConfig ksqlConfig;
+  @Mock(MockType.NICE)
+  private KsqlEngine mockKsqlEngine;
+  @Mock(MockType.NICE)
+  private KafkaTopicClient mockKafkaTopicClient;
+  @Mock(MockType.NICE)
+  private StatementParser mockStatementParser;
+  @Mock(MockType.NICE)
+  private ActivenessRegistrar activenessRegistrar;
+  private StreamedQueryResource testResource;
+
+  final String queryString = "SELECT * FROM test_stream;";
+
+  @Before
+  public void setup() {
     expect(mockKsqlEngine.getTopicClient()).andReturn(mockKafkaTopicClient);
+    expect(mockKsqlEngine.getLivePersistentQueries()).andReturn(Collections.emptySet());
 
-    final StatementParser mockStatementParser = mock(StatementParser.class);
-    expect(mockStatementParser.parseSingleStatement(queryString))
-        .andThrow(new IllegalArgumentException("some msg only the parser would use"));
+    replay(mockKsqlEngine);
 
-    replay(mockKsqlEngine, mockKafkaTopicClient, mockStatementParser);
+    testResource = new StreamedQueryResource(
+        ksqlConfig, mockKsqlEngine, mockStatementParser, DISCONNECT_CHECK_INTERVAL, activenessRegistrar);
 
-    final StreamedQueryResource testResource = new StreamedQueryResource(
-        ksqlConfig, mockKsqlEngine, mockStatementParser, DISCONNECT_CHECK_INTERVAL);
+  }
 
-    final Response response =
-        testResource.streamQuery(new KsqlRequest(queryString, Collections.emptyMap()));
+  @Test
+  public void shouldReturn400OnBadStatement() throws Exception {
+    // Given:
+    expect(mockStatementParser.parseSingleStatement(anyString()))
+        .andThrow(new IllegalArgumentException("some error message"));
+
+    replay( mockStatementParser);
+
+    // When:
+    final Response response = testResource.streamQuery(new KsqlRequest("query", Collections.emptyMap()));
+
+    // Then:
     assertThat(response.getStatus(), equalTo(Response.Status.BAD_REQUEST.getStatusCode()));
     assertThat(response.getEntity(), instanceOf(KsqlErrorMessage.class));
     final KsqlErrorMessage errorMessage = (KsqlErrorMessage)response.getEntity();
     assertThat(errorMessage.getErrorCode(), equalTo(Errors.ERROR_CODE_BAD_REQUEST));
-    assertThat(
-        errorMessage.getMessage(), containsString("some msg only the parser would use"));
+    assertThat(errorMessage.getMessage(), containsString("some error message"));
   }
 
   @Test
   public void shouldReturn400OnBuildMultipleQueriesError() throws Exception {
-    final String queryString = "SELECT * FROM test_stream;";
+    // Given:
+    expect(mockStatementParser.parseSingleStatement(anyString()))
```

Comment:
Going by the name of this test, it should be `buildMultipleQueries` that throws in this test, not `parseSingleStatement`...

Reply:
Of course it calls `parseSingleStatement`.  But the test used to throw from `buildMultipleQueries` and you've changed it to throw from `parseSingleStatement`. 

There is already a test which tests what happens when `parseSingleStatement` throws. It's the test above this one: `shouldReturn400OnBadStatement`.

Even the name of this test indicates its testing what happens when `buildMultipleQueries` throws.

Reply:
As we discussed `buildMultipleQueries` is not called in `StreamedQueryResource`. So this test indeed is not necessary anymore! I removed it.